pkg/operator/ceph/cluster/osd/topology.go

```diff
@@ -30,3 +41,46 @@ var (
 	// The list of supported failure domains in the CRUSH map, ordered from lowest to highest
 	CRUSHMapLevelsOrdered = append([]string{"host"}, append(CRUSHTopologyLabels, KubernetesTopologyLabels...)...)
 )
+
+// ExtractRookTopologyFromLabels extracts rook topology from labels and returns a map from topology type to value,
+// and an array of any invalid labels with a topology prefix.
+func ExtractRookTopologyFromLabels(labels map[string]string) (map[string]string, []string) {
+	topology := make(map[string]string)
+
+	// get zone
+	zone, ok := labels[corev1.LabelZoneFailureDomain]
+	if ok {
+		topology["zone"] = client.NormalizeCrushName(zone)
+	}
+	// get region
+	region, ok := labels[corev1.LabelZoneRegion]
+	if ok {
+		topology["region"] = client.NormalizeCrushName(region)
+	}
+
+	// get host
+	host, ok := labels[corev1.LabelHostname]
+	if ok {
+		topology["host"] = client.NormalizeCrushName(host)
+	}
+
+	invalidEncountered := make([]string, 0)
+	for labelKey, labelValue := range labels {
+		for _, validTopologyType := range CRUSHTopologyLabels {
+			if strings.HasPrefix(labelKey, k8sutil.TopologyLabelPrefix) {
+				s := strings.Split(labelKey, "/")
+				if len(s) != 2 {
+					invalidEncountered = append(invalidEncountered, fmt.Sprintf("%s=%s", labelKey, labelValue))
```

Comment:
Why not just log the warning here and continue? No need to return the list, right?

Reply:
Wanted to have this method as a resuable topology method. We are in fact using it in multiple places, so we'd end up with repeat logs. This way we can reuse it, and discard the info about invalid labels in some places.